packages/nimble-components/src/table-column/icon/models/table-column-icon-validator.ts

```diff
@@ -0,0 +1,44 @@
+import type { Mapping } from '../../../mapping/base';
+import { MappingIcon } from '../../../mapping/icon';
+import type { ColumnInternals } from '../../base/models/column-internals';
+import {
+    TableColumnEnumBaseValidator,
+    enumBaseValidityFlagNames
+} from '../../enum-base/models/table-column-enum-base-validator';
+import type { MappingKeyType } from '../../enum-base/types';
+
+const iconValidityFlagNames = [
+    ...enumBaseValidityFlagNames,
+    'invalidIconName'
+] as const;
+
+/**
+ * Validator for TableColumnIcon
+ */
+export class TableColumnIconValidator extends TableColumnEnumBaseValidator<
+    typeof iconValidityFlagNames
+> {
+    public constructor(
+        columnInternals: ColumnInternals<unknown>,
+        supportedMappingElements: readonly (typeof Mapping)[]
+    ) {
+        super(columnInternals, iconValidityFlagNames, supportedMappingElements);
+    }
+
+    public override validate(
+        mappings: Mapping[],
+        keyType: MappingKeyType
+    ): void {
+        super.validate(mappings, keyType);
+        this.validateIconNames(mappings);
+    }
+
+    private validateIconNames(mappings: Mapping[]): void {
+        const isMappingIcon = (mapping: Mapping): mapping is MappingIcon => mapping instanceof MappingIcon;
+        const invalid = mappings
+            .filter(isMappingIcon)
+            .map(mappingIcon => mappingIcon.resolvedIcon)
+            .some(resolvedIcon => resolvedIcon === undefined);
```

Comment:
nit: I don't expect there to be too many mapping icons, but it seems unnecessary to convert the array of MappingIcon instances to a string array before looking at the value of those strings. Why not do this:

```suggestion
        const invalid = mappings
            .filter(isMappingIcon)
            .some(mappingIcon => mappingIcon.resolvedIcon === undefined);
```